mne/io/edf/tests/test_edf.py

```diff
@@ -818,37 +892,40 @@ def test_empty_chars():
 def _hp_lp_rev(*args, **kwargs):
     out, orig_units = _read_edf_header(*args, **kwargs)
     out["lowpass"], out["highpass"] = out["highpass"], out["lowpass"]
-    # this will happen for test_edf_stim_resamp.edf
-    if (
-        len(out["lowpass"])
-        and out["lowpass"][0] == "0.000"
-        and len(out["highpass"])
-        and out["highpass"][0] == "0.0"
-    ):
-        out["highpass"][0] = "10.0"
```

Comment:
String `"0.000"` for lowpass was converted into float `0` in the previous code.
(it does not pass `if lowpass[0] in ("NaN", "0", "0.0"):`)
In this case, both lowpass and highpass is set to 0, and 
`if info["highpass"] > info["lowpass"]:`
is False.

I think this is a kind of special test for `highpass > lowpass` case.

With the new code, `"0.000"` for lowpass is skipped to be set as `info["lowpass"]` and it remains as default (`sfreq/2`).
Therefore, lowpass=256, highpass=0 are correctly obtained.

To make `highpass > lowpass` case, `_hp_lp_mod` function was added to force to `set highpass > lowpass`.